logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/DeleteMessageUseCase.kt

```diff
@@ -81,4 +64,20 @@ class DeleteMessageUseCase(
                 }
             }
     }
+
+    private suspend fun removeMessageAsset(conversationId: ConversationId, messageId: String) {
+        messageRepository.getMessageById(conversationId, messageId)
+            .onSuccess { message ->
+                (if (message.content is MessageContent.Asset)
+                    (message.content as MessageContent.Asset).value.remoteData else null)
```

Comment:
I think you can use safe cast operator `as?` instead 
`(messageToRemove.content as? MessageContent.Asset)?.value?.remoteData`

Reply:
Good idea 👍